code/modules/materials/_materials.dm

```diff
@@ -745,34 +741,33 @@ INITIALIZE_IMMEDIATE(/obj/effect/gas_overlay)
 				return
 		M.clean_blood()
 
-	if(solvent_power >= MAT_SOLVENT_STRONG && removed >= solvent_melt_dose)
-
-		if(ishuman(M))
-			var/mob/living/carbon/human/H = M
-			for(var/slot in global.standard_headgear_slots)
-				var/obj/item/thing = H.get_equipped_item(slot)
-				if(!istype(thing))
-					continue
-				if(thing.unacidable || !H.try_unequip(thing))
-					to_chat(H, SPAN_NOTICE("Your [thing] protects you from the acid."))
-					holder.remove_reagent(type, REAGENT_VOLUME(holder, type))
-					return
-				to_chat(H, SPAN_DANGER("Your [thing] dissolves!"))
-				qdel(thing)
-				removed -= solvent_melt_dose
-				if(removed <= 0)
-					return
-
-			if(!H.unacidable)
-				var/screamed
-				for(var/obj/item/organ/external/affecting in H.get_external_organs())
-					if(!screamed && affecting.can_feel_pain())
-						screamed = TRUE
-						H.emote("scream")
-					affecting.status |= ORGAN_DISFIGURED
-
-		if(!M.unacidable)
-			M.take_organ_damage(0, min(removed * solvent_power * ((removed < solvent_melt_dose) ? 0.1 : 0.2), solvent_max_damage), override_droplimb = DISMEMBER_METHOD_ACID)
+	if(solvent_power > MAT_SOLVENT_NONE && removed >= solvent_melt_dose && M.acid_act(min(removed * solvent_power * ((removed < solvent_melt_dose) ? 0.1 : 0.2), solvent_max_damage), solvent_melt_dose, solvent_power))
+		holder.remove_reagent(type, REAGENT_VOLUME(holder, type))
+
+/mob/living/proc/acid_act(var/severity, var/amount_per_item, var/solvent_power)
```

Comment:
shouldn't this be somewhere in mob code with similar damage procs, kinda weird having it in material module

Reply:
Personally I think it's fine to keep acid-related code in one section? We already do something similar with multiz code.